src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -227,10 +228,14 @@ object QueryService extends LazyLogging {
               tripleString = TripleString(source, predicate, target)
               originalKS <- ZIO.fromOption(biolinkData.predicates.find(p => p.shorthand == "original_knowledge_source")).orElseFail(new Exception("could not get biolink:original_knowledge_source"))
               aggregatorKS <- ZIO.fromOption(biolinkData.predicates.find(p => p.shorthand == "aggregator_knowledge_source")).orElseFail(new Exception("could not get biolink:aggregator_knowledge_source"))
+
               provValue <- ZIO.fromOption(provs.get(tripleString)).orElseFail(new Exception("no prov value"))
               infoResBiolinkClass <- ZIO.fromOption(biolinkData.classes.find(p => p.shorthand == "InformationResource")).orElseFail(new Exception("could not get biolink:InformationResource"))
               aggregatorKSAttribute = TRAPIAttribute(Some("infores:cam-kp"), aggregatorKS.iri, None, List("infores:cam-kp"), Some(infoResBiolinkClass.iri), Some(appConfig.location), None, None)
-              originalKSAttribute = TRAPIAttribute(Some("infores:cam-kp"), originalKS.iri, None, List("infores:go-cam"), Some(infoResBiolinkClass.iri), Some(provValue), None, None)
+              originalKSAttribute = provValue match {
+                case ctd if provValue.contains("ctdbase.org") => TRAPIAttribute(Some("infores:cam-kp"), originalKS.iri, None, List("infores:ctd"), Some(infoResBiolinkClass.iri), Some(provValue), None, None)
+                case _ => TRAPIAttribute(Some("infores:cam-kp"), originalKS.iri, None, List("infores:go-cam"), Some(infoResBiolinkClass.iri), Some(provValue), None, None)
+              }
```

Comment:
I think this would be simpler if you just have an if...else that stores the infores value in a string, then use it in a single `TRAPIAttribute` line like before.